File: Stack&Queue.cpp
#include <iostream>
#include <vector>
using namespace std;

// Stack
class Stack {
private:
    vector<int> stack;

public:
    void push(int item) {
        stack.push_back(item);
    }

    int pop() {
        if (!stack.empty()) {
            int item = stack.back();
            stack.pop_back();
            return item;
        }
        return -1; // Return -1 if stack is empty
    }

    int peek() {
        return stack.empty() ? -1 : stack.back();
    }

    bool is_empty() {
        return stack.empty();
    }
};

// Queue
class Queue {
private:
    vector<int> queue;

public:
    void enqueue(int item) {
        queue.push_back(item);
    }

    int dequeue() {
        if (!queue.empty()) {
            int item = queue.front();
            queue.erase(queue.begin());
            return item;
        }
        return -1; // Return -1 if queue is empty
    }

    int peek() {
        return queue.empty() ? -1 : queue.front();
    }

    bool is_empty() {
        return queue.empty();
    }
};

File: Stack&Queue.java
import java.util.LinkedList;

// Stack
class Stack {
    private LinkedList<Integer> stack = new LinkedList<>();

    public void push(int item) {
        stack.addLast(item);
    }

    public int pop() {
        return stack.isEmpty() ? -1 : stack.removeLast();
    }

    public int peek() {
        return stack.isEmpty() ? -1 : stack.getLast();
    }

    public boolean isEmpty() {
        return stack.isEmpty();
    }
}

// Queue
class Queue {
    private LinkedList<Integer> queue = new LinkedList<>();

    public void enqueue(int item) {
        queue.addLast(item);
    }

    public int dequeue() {
        return queue.isEmpty() ? -1 : queue.removeFirst();
    }

    public int peek() {
        return queue.isEmpty() ? -1 : queue.getFirst();
    }

    public boolean isEmpty() {
        return queue.isEmpty();
    }
}
